sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -710,39 +725,53 @@ synchronized void completeWorkflow() {
     /*
      * Store update token and possibly trigger application update check.
      */
-    synchronized void storeUpdateToken(@NonNull String updateToken, @NonNull String requestId) {
+    synchronized void storeRedirectionParameters(@NonNull String requestId, @NonNull String distributionGroupId, String updateToken) {
 
         /* Keep token for later if we are not started and enabled yet. */
         if (mContext == null) {
             MobileCenterLog.debug(LOG_TAG, "Update token received before onStart, keep it in memory.");
-            mBeforeStartUpdateToken = updateToken;
             mBeforeStartRequestId = requestId;
+            mBeforeStartUpdateToken = updateToken;
+            mBeforeStartDistributionGroupId = distributionGroupId;
         } else if (requestId.equals(PreferencesStorage.getString(PREFERENCE_KEY_REQUEST_ID))) {
-            String encryptedToken = CryptoUtils.getInstance(mContext).encrypt(updateToken);
-            PreferencesStorage.putString(PREFERENCE_KEY_UPDATE_TOKEN, encryptedToken);
+            if (updateToken != null) {
+                String encryptedToken = CryptoUtils.getInstance(mContext).encrypt(updateToken);
+                PreferencesStorage.putString(PREFERENCE_KEY_UPDATE_TOKEN, encryptedToken);
+            } else {
+                PreferencesStorage.remove(PREFERENCE_KEY_UPDATE_TOKEN);
```

Comment:
What I meant was this scenario.
1. An app has been installed from private distribution group.
2. The app opened and completed all install processes.
3. A newer version has been installed not from in-app-update but from install URL. -> This is still possible.
4. After opening the app, update token will be wiped out. -> We can prevent this case without the line and the app will still be able to get newer versions from private distribution group.